Sources/PT.PM.CSharpParseTreeUst/Converter/Helper.cs

```diff
@@ -35,6 +41,100 @@ protected TypeToken ConvertType(Ust node)
             return null;
         }
 
+        /// <summary>
+        /// Performs deconstruction of the tuples, in the absence of a name, else create vitual memrefs
+        /// </summary>
+        private Ust HandleTupleStatement(VariableDeclarationSyntax node)
+        {
+            List<AssignmentExpression> variables =
+                node.Variables.Select(v => (AssignmentExpression)VisitAndReturnNullIfError(v)).ToList();
+
+            var typeNode = (TupleTypeSyntax)node.Type;
+            var textSpan = node.GetTextSpan();
+
+            for (int i = 0; i < typeNode.Elements.Count; i++)
+            {
+                TupleElementSyntax typeElement = typeNode.Elements[i];
+                if (typeElement.Identifier == null)
+                {
+                    continue;
+                }
+
+                CreateVirtualMemRefs(ref variables, typeElement, i, textSpan);
+            }
+            var declaration = new VariableDeclarationExpression(null, variables, textSpan);
+            return new ExpressionStatement(declaration);
+        }
+
+        private void CreateVirtualMemRefs(ref List<AssignmentExpression> variables, TupleElementSyntax typeElement, int initializerNumber, TextSpan textSpan)
```

Comment:
I suggest renaming `MemRefs` to `MemberRefs` not to confuse with "memory ref".

Reply:
good idea, thanks